pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -222,8 +223,15 @@ async def add_tool(tool: Tool[DepsT]) -> None:
     async def add_mcp_server_tools(server: MCPServer) -> None:
         if not server.is_running:
             raise exceptions.UserError(f'MCP server is not running: {server}')
-        tool_defs = await server.list_tools()
-        # TODO(Marcelo): We should check if the tool names are unique. If not, we should raise an error.
+        tool_defs: list[ToolDefinition] = await server.list_tools()
+
+        tool_name_counts: Counter[str] = Counter(tool_def.name for tool_def in tool_defs)
```

Comment:
Can we also verify that the MCP server's tools don't clash with tools from _other_ servers, as well as the user-defined tools? We can look at all of the `function_tool_defs`. 

I suggest looking at the `Agent._register_tool` method that does this when adding a new user-defined tool.

Instead of using a Counter, I think we can turn `function_tool_defs` into a `dict`, and when iterating over the new tools, check if a key for that name already exists.